### PR TITLE
issue 3: cut over to opensearch

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Moved (cut over) to opensearch instead of elasticsearch. Some refactoring took place to support the move.

## ⚙️ Test Data and/or Report
Integration tests pass:

## ♻️ Related Issues
#3

### CODE REVIEW

File: service/src/main/java/gov/nasa/pds/api/registry/search/HitIterator.java [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
-import org.elasticsearch.search.SearchHit;
-import org.elasticsearch.search.SearchHits;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.RestHighLevelClient;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.SearchHits;
 
-public class ElasticSearchHitIterator implements Iterable<Map<String,Object>>,Iterator<Map<String,Object>>
+public class HitIterator implements Iterable<Map<String,Object>>,Iterator<Map<String,Object>>
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/search/HitIterator.java:13]: ignore

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public ElasticSearchRegistryConnectionImpl(List<String> hosts,
 		if ((username != null) && (username != ""))  {
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/registry/business/LidVidUtils.java [hunk 3]
```diff
@@ -59,7 +59,7 @@ public static String extractLidFromLidVid(String identifier)
      * @return list of LIDVIDs
      * @throws IOException
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/business/LidVidUtils.java:60]: ignore

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java [hunk 4]
```diff
@@ -58,10 +58,10 @@ public ElasticSearchRegistryConnectionImpl(List<String> hosts,
 		
 		List<HttpHost> httpHosts = new ArrayList<HttpHost>();
 		
-		ElasticSearchRegistryConnectionImpl.log.info("Connection to elastic search");
+		OpenSearchRegistryConnectionImpl.log.info("Connection to elastic search");
 		for (String host : hosts) {
 			String hostPort[] = host.split(":");
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java:63]: ignore

File: service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java [hunk 5]
```diff
@@ -136,7 +136,7 @@ private static XMLMashallableProperyValue object2PropertyValue(Object o) {
                 String apiProperty;
                 for (Map.Entry<String, Object> entry : sourceAsMap.entrySet()) {
                     try {
-                        apiProperty = ElasticSearchUtil.elasticPropertyToJsonProperty(entry.getKey());
+                        apiProperty = SearchUtil.elasticPropertyToJsonProperty(entry.getKey());
                         if ((excluded_fields == null)
                                 || (! excluded_fields.contains(apiProperty)))
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java [hunk 6]
```diff
@@ -119,7 +119,7 @@ private static XMLMashallableProperyValue object2PropertyValue(Object o) {
        
        
        /**
-     * @param sourceAsMap source map coming from elasticSearch
+     * @param sourceAsMap source map coming from openSearch
      * @param included_fields, in API syntax, with .
      * @param excluded_fields is ignored is included_fields is not null and not empty, in API syntax
      * @return
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java:125]: ignore

File: service/src/main/java/gov/nasa/pds/api/registry/search/RegistrySearchRequestBuilder.java [hunk 7]
```diff
@@ -7,25 +7,25 @@
...
+    private static final Logger log = LoggerFactory.getLogger(RegistrySearchRequestBuilder.class);
     
     final public String registryIndex;
     final public String registryRefIndex;
     private int timeOutSeconds;
     
-    public ElasticSearchRegistrySearchRequestBuilder(
+    public RegistrySearchRequestBuilder(
             String registryIndex, 
             String registryRefindex,
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/search/RegistrySearchRequestBuilder.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java [hunk 8]
```diff
@@ -119,7 +119,7 @@ private static XMLMashallableProperyValue object2PropertyValue(Object o) {
        
        
        /**
-     * @param sourceAsMap source map coming from elasticSearch
+     * @param sourceAsMap source map coming from openSearch
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/business/ProductBusinessObject.java:122]: ignore

**Comment by al-niessner** [comment]
@jordanpadams @tloubrieu-jpl 

It now uses opensearch instead of elasticsearch. According to find, all elastic have been excised from the code.

Can we do a high priority fix to match the documentation (swagger.yml) with code? I would like to keep repo frozen after this branch merges in for the high priority fix. I should be done with it by Thursday but will be sending lots of emails for details to put in docs when I find discrepancies.

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java [hunk 9]
```diff
@@ -70,10 +70,10 @@ public ElasticSearchRegistryConnectionImpl(List<String> hosts,
 		
 		RestClientBuilder builder;
 		
-		if ((username != null) && (username != ""))  {
+		if ((username != null) && (!username.equals("")))  {
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java]: fixed

**Comment by al-niessner** [comment]
@jordanpadams @tloubrieu-jpl 

Can I get this reviewed so I can merge it please?

**Comment by tloubrieu-jpl** [comment]
@al-niessner sorry about the delay here, I am busy on a different project today and yesterday I was blocked by something not working right with the docker-compose deployment (which I thought could help me).

To unblock you you can create a branch off this branch.

**Comment by al-niessner** [comment]
Seems I can wait. I am blocked by gateway errors at swaggerhub more than this and there is nothing any of us can do about swaggerhub but wait.

**Review by tloubrieu-jpl — APPROVED** [review]
Thanks @al-niessner , that works

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

This PR is closed but not merged. Is there a reason for that?

**Comment by tloubrieu-jpl** [comment]
@al-niessner my bad, it was merged on my local repository (do you care ?) and not on github. And the pull request had a tag "merged" already, I don't understand why, maybe it wads only partially merged...

**Comment by al-niessner** [comment]
Yes, I am trying to get my next round of changes back to main and do not want a giant conflict so I would like this PR merged. Since it is in your repo then all you need do it push main and we should be good.

**Comment by jordanpadams** [comment]
@al-niessner your issue_3.1 branch was merged with main per above. can you check that you have the latest changes on main and make your updates on a new branch from there?

i can confirm your commits were merged to main per e824e39813ac785d1609872b632702491143effe , so unless someone removed them in a later commit, that should be done.

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

Yes it is there now. Thanks.

**Pull request merged**